custom_components/renault/binary_sensor.py

```diff
@@ -51,11 +51,11 @@ class RenaultPluggedInSensor(RenaultBatteryDataEntity, BinarySensorEntity):
     @property
     def is_on(self) -> Optional[bool]:
         """Return true if the binary sensor is on."""
-        return (
-            self.data.get_plug_status() == PlugState.PLUGGED
-            if self.data.plugStatus is not None
-            else None
-        )
+        if not self.data:
+            return None
+        if self.data.plugStatus is None:
+            return None
```

Comment:
What do you think about this simplification ?
```suggestion
        if (not self.data) or (self.data.plugStatus is None):
            return None
```
the `or`operator will not evaluate the second operant if the first one is True